hs_core/hydroshare/resource.py

```diff
@@ -559,13 +559,20 @@ def create_new_version_resource(ori_res, new_res):
         res_avail_date = new_res.metadata.dates.all().filter(type='available')[0]
         new_res.metadata.create_element('date', type='available', start_date=res_avail_date.start_date, end_date=res_avail_date.end_date)
 
-    # add Relation element to link source and target resources
+    # add or update Relation element to link source and target resources
     if new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier"):
         hs_identifier = new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
         ori_res.metadata.create_element('relation', type='isReplacedBy', value=hs_identifier.url)
     else:
         ori_res.metadata.create_element('relation', type='isReplacedBy', value=new_res.short_id)
```

Comment:
@pkdash I believe so, since we have to add an "isReplacedBy" element for a published resource if we allow it to be versioned. I guess we don't allow users to modify published resources in any way, but HydroShare system can modify published resources' metadata as needed, e.g., when a new version for the published resource is created. @dtarb @horsburgh what do you think? I also want to point out since we don't show obsoleted resources on My Resources listing, even if obsoleted resources are published, they will be replaced with their new version counterparts on My Resources listing.